### PR TITLE
Add config for disabling hover memory layout data

### PR DESCRIPTION
**Pull request by lumenian**
Requested in https://github.com/rust-lang/rust-analyzer/pull/14748#issuecomment-1537190252

### CODE REVIEW

File: crates/rust-analyzer/src/config.rs [hunk 1]
```diff
@@ -313,8 +313,10 @@ config_data! {
         /// Whether to show keyword hover popups. Only applies when
         /// `#rust-analyzer.hover.documentation.enable#` is set.
         hover_documentation_keywords_enable: bool  = "true",
-        /// Use markdown syntax for links in hover.
+        /// Use markdown syntax for links on hover.
         hover_links_enable: bool = "true",
+        /// Whether to show memory layout data on hover.
+        hover_memory_layout_enable: bool = "true",
```
Comment (Veykril) [inline crates/rust-analyzer/src/config.rs:319]:
```suggestion
        hover_memoryLayout_enable: bool = "true",
```

Let's treat this as one thing

Comment (lumenian) [reply crates/rust-analyzer/src/config.rs]: Makes sense, thank you. Fixed it.

File: crates/ide/src/hover/render.rs [hunk 2]
```diff
@@ -577,17 +577,17 @@ where
...
     D: HasAttrs + HirDisplay,
     E: Fn(&D) -> Option<V>,
     V: Display,
 {
-    let label = if let Some(value) = value_extractor(&def) {
-        format!("{} // {value}", def.display(db))
-    } else {
-        def.display(db).to_string()
+    let label = match value_extractor(&def) {
+        Some(value) if config.memory_layout => format!("{} // {value}", def.display(db)),
```
Comment (HKalbasi) [inline crates/ide/src/hover/render.rs:589]: Unfortunately closure hover doesn't use `label_and_layout_info_and_docs`, but it shows some layout info which needs to be disabled.

Comment (lumenian) [reply crates/ide/src/hover/render.rs:593]: Thank you, I missed that closures use a different implementation.
I've added an additional test for closures.

**Pull request merged**